src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirDataStore.cs

```diff
@@ -270,12 +282,13 @@ internal class SqlServerFhirDataStore : IFhirDataStore, IProvideCapability
                 resource.ResourceSurrogateId = surrId;
                 mergeWrappers.Add(new MergeResourceWrapper(resource, resourceExt.KeepHistory, hasVersionToCompare));
                 index++;
-                results.Add(resourceKey, new UpsertOutcome(resource, resource.Version == InitialVersion ? SaveOutcomeType.Created : SaveOutcomeType.Updated));
+                results.Add(identifier, new DataStoreOperationOutcome(new UpsertOutcome(resource, resource.Version == InitialVersion ? SaveOutcomeType.Created : SaveOutcomeType.Updated)));
             }
 
             if (mergeWrappers.Count > 0) // do not call db with empty input
             {
-                using var conn = await _sqlConnectionWrapperFactory.ObtainSqlConnectionWrapperAsync(cancellationToken, true); // TODO: Remove tran enlist when true bundle logic is in place.
+                // TODO: Remove tran enlist when true bundle logic is in place.
+                using var conn = await _sqlConnectionWrapperFactory.ObtainSqlConnectionWrapperAsync(cancellationToken, enlistInTransaction: true);
```

Comment:
As discussed offline, "enlistInTransaction" will be changed in the next sprint when implementing the changes required to support bundle transactions running in parallel and it'll not be included in this PR to avoid increasing its size.